presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/TimestampOperators.java

```diff
@@ -0,0 +1,428 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar.timestamp;
+
+import io.airlift.slice.XxHash64;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.function.IsNull;
+import io.prestosql.spi.function.LiteralParameter;
+import io.prestosql.spi.function.LiteralParameters;
+import io.prestosql.spi.function.ScalarOperator;
+import io.prestosql.spi.function.SqlNullable;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.LongTimestamp;
+import io.prestosql.spi.type.LongTimestampType;
+import io.prestosql.spi.type.ShortTimestampType;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.type.Constraint;
+import org.joda.time.DateTimeField;
+import org.joda.time.chrono.ISOChronology;
+
+import static io.prestosql.spi.function.OperatorType.ADD;
+import static io.prestosql.spi.function.OperatorType.EQUAL;
+import static io.prestosql.spi.function.OperatorType.GREATER_THAN;
+import static io.prestosql.spi.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static io.prestosql.spi.function.OperatorType.HASH_CODE;
+import static io.prestosql.spi.function.OperatorType.INDETERMINATE;
+import static io.prestosql.spi.function.OperatorType.LESS_THAN;
+import static io.prestosql.spi.function.OperatorType.LESS_THAN_OR_EQUAL;
+import static io.prestosql.spi.function.OperatorType.NOT_EQUAL;
+import static io.prestosql.spi.function.OperatorType.SUBTRACT;
+import static io.prestosql.spi.function.OperatorType.XX_HASH_64;
+import static io.prestosql.type.Timestamps.MICROSECONDS_PER_MILLISECOND;
+import static io.prestosql.type.Timestamps.getMicrosOfMilli;
+import static io.prestosql.type.Timestamps.rescale;
+import static io.prestosql.type.Timestamps.scaleEpochMicrosToMillis;
+import static io.prestosql.type.Timestamps.scaleEpochMillisToMicros;
+import static io.prestosql.util.DateTimeZoneIndex.getChronology;
+import static java.lang.Math.multiplyExact;
+
+public final class TimestampOperators
+{
+    private TimestampOperators() {}
+
+    @ScalarOperator(EQUAL)
+    public static final class Equal
+    {
+        @LiteralParameters("p")
+        @SqlNullable
+        @SqlType(StandardTypes.BOOLEAN)
+        public static Boolean equal(@SqlType("timestamp(p)") long left, @SqlType("timestamp(p)") long right)
+        {
+            return left == right;
+        }
+
+        @LiteralParameters("p")
+        @SqlNullable
+        @SqlType(StandardTypes.BOOLEAN)
+        public static Boolean equal(@SqlType("timestamp(p)") LongTimestamp left, @SqlType("timestamp(p)") LongTimestamp right)
+        {
+            return left.equals(right);
+        }
+    }
+
+    @ScalarOperator(NOT_EQUAL)
+    public static final class NotEqual
```

Comment:
nit: `Class 'LessThan' has only 'static' members, and lacks a 'private' constructor`

Reply:
I think that's too noisy. We end up with a bunch of dummy constructors for no reason. It's clear that these classes are not meant to be used or instantiated.

Reply:
You can suppress this on top-level class level. Ignoreable IDE warning make it harder to notice the ones i shouldn't ignore.